astropy/io/misc/ecsv.py

```diff
@@ -74,6 +75,8 @@
 
 __all__ = ["read_ecsv", "register_ecsv_table", "write_ecsv"]
 
+ECSVEngines: dict[str, "ECSVEngine"] = {}
```

Comment:
We also discussed using a [WeakValueDictionary](https://docs.python.org/3/library/weakref.html#weakref.WeakValueDictionary)